Automattic/wp-calypso
# Calypso Masterbar: add mini shopping cart, take 2

#### Changes proposed in this Pull Request

This is a second attempt at #58467 which was reverted because the mini-cart was using the site slug as the cart key for the shopping-cart manager, but the rest of calypso uses the site ID for this purpose. As a result, the cart manager thought there were two different carts. 

In many cases this difference is not important because the server will treat them as the same and will return the same data, but if you modify the cart (eg: adding items or removing them), the cart manager for the site slug version of the cart _won't be updated_ until you reload the page (or otherwise trigger a fetch like by refocusing the window after a delay).

That issue can make it appear that you have items in your cart when you do not, or that you don't have items in your cart when you do.

In this PR, we change mini-cart to use the site ID for the cart key instead.

<img width="781" alt="146074547-a57bd615-5f51-40c0-a326-69a7b389a1a8" src="https://user-images.githubusercontent.com/2036909/148130843-c48cd2f2-e5c7-4b4c-ab2c-f00de90d6f9a.png">


#### Testing instructions

- Add a product to your cart and visit checkout without reloading the page or unfocusing it.
- Remove the cart item and exit checkout without reloading or unfocusing.
- Verify that the masterbar cart does not show up.
- Add a product to your cart again.
- Leave checkout without emptying the cart (use the X button and click "Leave items").
- Verify that the masterbar cart does show up.
